atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransaction.java

```diff
@@ -657,6 +731,65 @@ private void verifyColumnRanges(Transaction readOnlyTransaction) {
         }
     }
 
+    private void verifyGetSortedColumns(Transaction readOnlyTransaction) {
+        sortedColumnRangeEnds.forEach((request, endOfRangeReference) -> {
+            Cell endOfRange = endOfRangeReference.get();
+            // no checks required if no data has been read so far
+            if (endOfRange == null) {
+                return;
+            }
+            BatchColumnRangeSelection range =
+                    nextLexicographicalRangeEnd(request.getColumnRangeSelection(), endOfRange.getColumnName());
```

Comment:
A bit concerned about this if the end of range is genuinely the last possible column name, and there exists a value there. To be clear the conceptual operation is correct, just that `nextLexicographicalRangeEnd` itself might be buggy